core/templates/pages/story-editor-page/modal-templates/story-editor-unpublish-modal.component.spec.ts

```diff
@@ -63,4 +79,18 @@ describe('Story Editor Unpublish Modal Component', () => {
     component.confirm();
     expect(confirmSpy).toHaveBeenCalled();
   });
+
+  it('should set unpublishing reason', () => {
+    mockPlatformFeatureService.
+      status.SerialChapterLaunchCurriculumAdminView.isEnabled = true;
+
+    component.setReason('BAD_CONTENT');
+    expect(component.unpublishingReason).toBe('BAD_CONTENT');
+    expect(component.selectedReasonText).toBe(component.badContentReasonText);
```

Comment:
Add a pre-check for this test.

Reply:
Done

Reply:
@Kej-r03 why are we setting the `unpublishingReason` as `BAD_CONTENT` if it is already set to `BAD_CONTENT`? Can we test with a different value before `setReason` is called

Reply:
@gp201 The default value of unpublishingReason is BAD_CONTENT before calling setReason(). If this seems redundant maybe I can remove this unit test, and test for setReason('Bad_CONTENT') in the next unit test after checking 'CHAPTER_NEEDS_SPLITTING'. Does that sound fine?

Reply:
@Kej-r03 Instead, can you set the `unpublishingReason` to any other value before executing `component.setReason('BAD_CONTENT');`?

Reply:
@Kej-r03 Regarding this [comment](https://github.com/oppia/oppia/pull/18336#issuecomment-1591775837). We write the tests from a user perspective. Ideally, it would be better to have a placeholder. But let's test the scenario where we check if bad content is the default value as the test. Ignore the above comment and re-write the test.